do/sandbox.go

```diff
@@ -1,23 +1,73 @@
 package do
 
 import (
+	"context"
 	"encoding/json"
 	"errors"
+	"net/http"
 	"os"
 	"os/exec"
+	"strings"
+
+	"github.com/digitalocean/godo"
 )
 
-// SandboxService is an interface for interacting with the sandbox plugin.
+// SandboxCredentials is the type returned by the ResolveToken and ResolveNamespace functions
+// The values in it can be used to connect sandbox support to a specific namespace using the plugin.
+type SandboxCredentials struct {
+	Auth    string
+	ApiHost string
+}
+
+// The type of the "namespace" member of POST input for API calls.
+// Only one of the fields is typically specified
+type inputNamespace struct {
+	Token     string `json:"token,omitempty"`
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// The type of the "namespace" member of the response to API calls.  Only some
+// fields are relevant to each call.
+type outputNamespace struct {
+	ApiHost   string `json:"api_host"`
+	Uuid      string `json:"uuid"`
+	Key       string `json:"key"`
+	Token     string `json:"token"`
+	Label     string `json:"label"`
+	Namespace string `json:"namespace"`
+}
+
+// postBody is the type of the request body for v2/function/namespaces/... calls that use
+// the POST verb.  Only the relevant parts of the contained inputNamespace need be specified
+type postBody struct {
```

Comment:
A lot of this will eventually want to be in `godo` For now, since these types are shared across the package, it would be great to namespaces them (no pun intended). E.g.

```suggestion
type namespacePostBody struct {
```